kolibri/plugins/style_guide/assets/src/views/content/api_table.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+
+  <div>
+
+    <h3>Name: {{ api.name }}</h3>
+    <h3>Description: {{ api.description }}</h3>
+
+    <div v-if="api.props.length">
+      <h3>Props</h3>
+      <table>
+      <tr>
+        <th>Name</th>
+        <th>Type</th> 
+        <th>Required</th>
+        <th>Default</th>
+        <th>Description</th>
+      </tr>
+      <tr v-for="prop in api.props">
+        <td>{{ prop.name }}</td>
+        <td>{{ prop.value.type }}</td>
+        <td>{{ prop.value.required }}</td>
+        <td>{{ prop.value.default ? prop.value.default : 'None' }}</td>
+        <td>{{ prop.description }}</td>
+      </tr>
+      </table>
+    </div>
+
+    <div v-if="api.events.length">
+      <h3>Emitted events</h3>
+      <table>
+      <tr>
+        <th>Name</th>
+        <th>Type</th> 
+        <th>Default</th>
+        <th>Description</th>
+      </tr>
+      <tr v-for="event in api.events">
+        <td>{{ event.name }}</td>
+        <td>{{ event.type }}</td>
+        <td>{{ event.default ? event.default : 'None' }}</td>
+        <td>{{ event.description }}</td>
+      </tr>
+      </table>
+    </div>
+
+    <div v-if="api.slots.length">
+      <h3>Slots</h3>
+      <table>
+      <tr>
+        <th>Name</th>
+        <th>Description</th>
+      </tr>
+      <tr v-for="slot in api.slots">
+        <td>{{ slot.name }}</td>
+        <td>{{ slot.description }}</td>
+      </tr>
+      </table>
+    </div>
+
+    <div v-if="api.methods.length">
+      <h3>Methods</h3>
+      <table>
+      <tr>
+        <th>Name</th>
+        <th>Description</th>
+      </tr>
+      <tr v-for="method in api.methods">
+        <td>{{ method.name }}</td>
+        <td>{{ method.description }}</td>
+      </tr>
+      </table>
+    </div>
+
+  </div>
+
+</template>
+
+
+<script>
+
+  module.exports = {
+    props: {
+      api: {
+        type: Object,
```

Comment:
Some documentation of what form this object takes would be helpful. Is it specified by the output of the vuedoc parser?

A validation function would be especially awesome, but probably overkill.